File: 100-Days-Series/Day 55/Butterfly.js
const printButterfly = (n) => {
    // Upper part of the butterfly
    for (let i = 1; i <= n; i++) {
        let line = "";
        for (let j = 1; j <= i; j++) {
            line += "*"; // Left stars
        }
        for (let j = 1; j <= 2 * (n - i); j++) {
            line += " "; // Spaces
        }
        for (let j = 1; j <= i; j++) {
            line += "*"; // Right stars
        }
        console.log(line);
    }

    // Lower part of the butterfly
    for (let i = n; i >= 1; i--) {
        let line = "";
        for (let j = 1; j <= i; j++) {
            line += "*"; // Left stars
        }
        for (let j = 1; j <= 2 * (n - i); j++) {
            line += " "; // Spaces
        }
        for (let j = 1; j <= i; j++) {
            line += "*"; // Right stars
        }
        console.log(line);
    }
};

// Example usage
printButterfly(4);

File: 100-Days-Series/Day 55/Butterfly.py
def print_butterfly(n):
    # Upper part of the butterfly
    for i in range(1, n + 1):
        print("*" * i, end="")  # Left stars
        print(" " * (2 * (n - i)), end="")  # Spaces
        print("*" * i)  # Right stars

    # Lower part of the butterfly
    for i in range(n, 0, -1):
        print("*" * i, end="")  # Left stars
        print(" " * (2 * (n - i)), end="")  # Spaces
        print("*" * i)  # Right stars


row = int(input("Enter the number of rows for the butterfly: "))
print_butterfly(row)
